CourseSketchProjects/CourseSketchDatabaseServer/src/main/java/database/institution/mongo/MongoInstitution.java

```diff
@@ -579,32 +580,35 @@ public static String hashUserId(final String userId, final String courseId) thro
     }
 
     @Override
-    public List<ProtoGrade> getAllAssignmentGradesInstructor(final String courseId, final String userId)
+    public List<ProtoGrade> getAllAssignmentGradesInstructor(final String courseId, final String authId)
             throws AuthenticationException, DatabaseAccessException {
-        return GradeManager.getAllAssignmentGradesInstructor(auth, database, courseId, userId);
+        return GradeManager.getAllAssignmentGradesInstructor(auth, database, courseId, authId);
     }
 
     @Override
-    public List<ProtoGrade> getAllAssignmentGradesStudent(final String courseId, final String userId)
+    public List<ProtoGrade> getAllAssignmentGradesStudent(final String courseId, final String authId)
             throws AuthenticationException, DatabaseAccessException {
-        return GradeManager.getAllAssignmentGradesStudent(auth, database, courseId, userId);
+        return GradeManager.getAllAssignmentGradesStudent(auth, database, courseId, authId, authId);
     }
 
     @Override
-    public void addGrade(final String adderId, final ProtoGrade grade) throws AuthenticationException, DatabaseAccessException {
-        GradeManager.addGrade(auth, database, adderId, grade);
+    public void addGrade(final String authId, final ProtoGrade grade) throws AuthenticationException, DatabaseAccessException {
+        GradeManager.addGrade(auth, database, authId, grade);
     }
 
     @Override
-    public ProtoGrade getGrade(final String requesterId, final String userId, final String courseId, final String assignmentId,
-            final String problemId) throws AuthenticationException, DatabaseAccessException {
-        return GradeManager.getGrade(auth, database, requesterId, userId, courseId, assignmentId, problemId);
+    public ProtoGrade getGrade(final String authId, final ProtoGrade gradeData) throws AuthenticationException, DatabaseAccessException {
+        return GradeManager.getGrade(auth, database, authId, authId, gradeData);
```

Comment:
so funny enough it is correct for this specific branch.  BUT in the later branch this actually a mistake so it is a good thing you found this!